### ISSUE TITLE
Different behavior between . = . + and shorthand . += append assign operator

### ISSUE THREAD

**Issue by drdaeman** [mikefarah/yq#1174]
**Describe the bug**
I've discovered that [contrary to the documentation](https://mikefarah.gitbook.io/yq/operators/add) it seems that yq has somewhat different behavior between `. += something` and `. = . + something`. The shorthand operator uses last RHS value everywhere, even if this value should be different across the modified nodes.

Version of yq: 4.23.1
Operating system: Debian GNU/Linux 11 (bullseye)
Installed via: binary release, downloaded from GitHub

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data.yml:
```yaml
items:
  foo:
    is: foo
  bar:
    is: bar
  baz:
    is: baz
```

**Command**
The command you ran:
```
yq eval 'with(.items[]; . += {"key": . | key})' data.yml
```
Note that `with(.items[]; . = . + {"key": . | key})` works, producing the expected result.

Also, I totally realize that this is a weird expression that should've been just `.key = key` - this is unfortunate result of distilling the issue straight to the point. In real-world situation I have a complex data structure where I'm appending an item to an array nested inside a map with part of this item dependent on the top-level map key.

**Actual behavior**

```yaml
items:
  foo:
    is: foo
    key: baz
  bar:
    is: bar
    key: baz
  baz:
    is: baz
    key: baz
```

**Expected behavior**

```yaml
items:
  foo:
    is: foo
    key: foo
  bar:
    is: bar
    key: bar
  baz:
    is: baz
    key: baz
```

And last but not least - thank you for a very useful tool! I've recently discovered yq when I was looking for something to preprocess huge CI configuration document and yq is truly a lifesaver.

**Comment by mikefarah** [comment]
Yep - that's definitely a bug, thanks for the clear example.

**Comment by mikefarah** [comment]
Fixed in 4.24.5
